app/views/devise/registrations/new.html.slim

```diff
@@ -1,27 +1,28 @@
-div.content-container
-  div.row
-    div.col-md-4.col-md-offset-4
-      h2
-        | Sign up
-      = form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f|
-        = devise_error_messages!
-        .field
-          = f.label :email
-          br
-          = f.email_field :email, autofocus: true
-        .field
-          = f.label :password
-          - if @minimum_password_length
-            em
-              | (
-              = @minimum_password_length
-              |  characters minimum)
-          br
-          = f.password_field :password, autocomplete: "off"
-        .field
-          = f.label :password_confirmation
-          br
-          = f.password_field :password_confirmation, autocomplete: "off"
-        .actions
-          = f.submit "Sign up"
-      = render "devise/shared/links"
+-if GalleryConfig.registration_allowed == true && GalleryConfig.username_login_allowed == true
```

Comment:
@mcrutch, When I turn the `registration_allowed` to `false` in the configs/settings, when going to the url: `SITE/users/sign_up`, the page isn't throwing a 404 or a page with an error on it, just showing up as a blank page. Nothing is exposed in the inspect element for this and directing to this page and the confirmation new, but just war curious if we wanted to do that or keep it as is?

Reply:
I tried something similar.  I set `username_login_allowed: false` and `registration_allowed: true` - does that even make sense as a valid configuration?  The views seemed weird at first (basically nothing except the github oauth button), but maybe it's actually doing the right thing.  Maybe it at least needs some words about how to get an account in that scenario?

Reply:
But yes other than that it looks ok to me.

Reply:
We could add some placeholder text to those views, but since those routes are built into devise it can get a little complicated to disable them in a more "error page" kind of way based on some of the things I saw on the web.

username_login_allowed:false automatically blocks out registration too.  Anyone with an account on your oauth provider can get in that way even if they don't already have an account but that was kind of my plan with this and it would be made clear in the docs.  Prevent people from signing up/logging in without using oauth, or just prevent random people from signing up for accounts and have it all done by admins on the backend. (The latter is the more likely scenario we are going to need soon)

Reply:
> username_login_allowed:false automatically blocks out registration too

Ok I thought that might be the case - makes sense.  I wouldn't worry too much about the views then.

Reply:
@mcrutch apologies, got distracted with the orgs ticket, will make the changes asap tomorrow for the alerts on those blank pages, then will approve it.

Reply:
@manfromjupyter I am making a few tweaks...since most of them render devise/shared/links at the bottom, as long as you have an oauth provider enabled something will show up (which would be the intended behavior for username_login disabled.  for sign_up I am tweaking just a bit.